src/tracer/MARBL_forcing_mod.F90

```diff
@@ -0,0 +1,378 @@
+!> This module provides a common datatype to provide forcing for MARBL tracers
+!! regardless of driver
+module MARBL_forcing_mod
+
+!! This module exists to house code used by multiple drivers in config_src/
+!! for passing forcing fields to MARBL
+!! (This comment can go in the wiki on the NCAR fork?)
+
+use MOM_diag_mediator,        only : safe_alloc_ptr, diag_ctrl, register_diag_field, post_data
+use MOM_time_manager,         only : time_type
+use MOM_error_handler,        only : MOM_error, WARNING, FATAL
+use MOM_file_parser,          only : get_param, log_param, param_file_type
+use MOM_grid,                 only : ocean_grid_type
+use MOM_unit_scaling,         only : unit_scale_type
+use MOM_interpolate,          only : external_field, init_external_field, time_interp_external
+use MOM_io,                   only : slasher
+use marbl_constants_mod,      only : molw_Fe
+use MOM_forcing_type,         only : forcing
+
+implicit none ; private
+
+#include <MOM_memory.h>
+
+public :: MARBL_forcing_init
+public :: convert_driver_fields_to_forcings
+
+!> Data type used to store diagnostic index returned from register_diag_field()
+!! For the forcing fields that can be written via post_data()
+type, private :: marbl_forcing_diag_ids
+  integer :: atm_fine_dust   !< Atmospheric fine dust component of dust_flux
+  integer :: atm_coarse_dust !< Atmospheric coarse dust component of dust_flux
+  integer :: atm_bc          !< Atmospheric black carbon component of iron_flux
+  integer :: ice_dust        !< Sea-ice dust component of dust_flux
+  integer :: ice_bc          !< Sea-ice black carbon component of iron_flux
+end type marbl_forcing_diag_ids
+
+!> Control structure for this module
+type, public :: marbl_forcing_CS
+  type(diag_ctrl), pointer :: diag => NULL() !< A structure that is used to
+                                             !! regulate the timing of diagnostic output.
+
+  real    :: dust_ratio_thres               !< coarse/fine dust ratio threshold
```

Comment:
Please consider adding the units to the descriptions of `dust_ratio_thres` and the next 9 real variables.

Also, would it be possible to make the contents of `marbl_forcing_CS` private, as is common for most other MOM6 control structures?  If a few of the parameters are needed elsewhere, one common approach used elsewhere in the MOM6 code would be to add an object-orient extractor function.

Reply:
First comment was addressed in 5402ce0; still need to look into making contents of `marbl_forcing_CS` private (I think it should be doable)

Reply:
rest of comment addressed in a2193cf